pyccel/codegen/printing/cwrappercode.py

```diff
@@ -119,7 +119,7 @@ def _print_FunctionDef(self, expr):
                 wrapper_body.append(cast_func(a , collect_var))
             else :
                 res_args.append(a)
-            build_keys += pytype_registry[a.dtype]
+            build_keys += pytype_parse_registry [a.dtype]
```

Comment:
I have called it pytype_**parse**_registry because it should be different from the pytype_build_registry. The code `p` used for collecting booleans doesn't exist for pybuild so as a minimum this value will have to be changed

Reply:
The functions responsible for returning the variable type and the cast function should be enough to make sure what type we are trying to build or parse.

Example for boolean type parsing is done with 'p' and building is done with 'O' so when parsing we return (bool , cast_pyint_to_bool) and when we are building value  return (pyobject, cast_bool_to_pybool)

Reply:
Ah ok I see. 👍